pkg/services/accesscontrol/resourcepermissions/service.go

```diff
@@ -77,9 +79,36 @@ func New(cfg *setting.Cfg,
 		actions = append(actions, action)
 	}
 
+	// create the store for actionsets
+	log := log.New("accesscontrol.resourcepermissions")
+	actionSetsStore := NewInMemoryActionSets(log)
+
+	for permission, actions := range options.PermissionsToActions {
+		// for actionsets we need to know the resource
+		resource := ""
+		for _, a := range actions {
+
+			// assuming we get the ordering of this right,
+			// such as "dashboards", "folders", "teams"
+			if strings.Contains(a, ":") {
+				resource = strings.Split(a, ":")[0]
+				if resource != "folders" {
+					// currently only whitelisting folders
+					continue
+				}
+				break
+			}
+		}
+		// PermissionsToActions: map[string][]string{
+		// 	"Edit":  ServiceAccountEditActions,
+		// 	"Admin": ServiceAccountAdminActions,
+		// },
+		actionSetsStore.StoreActionSet(resource, permission, actions)
```

Comment:
Why not do this in the loop above ([here](https://github.com/grafana/grafana/pull/86108/files#diff-df31fc307ddb09b17b286316a8cc9b4a866578297cbc60f090ec1c3a065749a5R65-R70))?

Reply:
thought of separating the logic for actionsets and it's implementation into a block vs having code for actionsets within the loop where we set the "original" permissions (that will eventually, be replaced).